domain/api/insert_tile.go

```diff
@@ -36,23 +27,13 @@ func insertTileHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "failed to decode body", http.StatusInternalServerError)
 	}
 
-	var err error = nil
-	switch requestBody.Direction {
-	case DirectionTop:
-		err = requestBody.Board.InsertTileTopAt(requestBody.Index)
-	case DirectionRight:
-		err = requestBody.Board.InsertTileRightAt(requestBody.Index)
-	case DirectionBottom:
-		err = requestBody.Board.InsertTileBottomAt(requestBody.Index)
-	case DirectionLeft:
-		err = requestBody.Board.InsertTileLeftAt(requestBody.Index)
-	default:
+	actions := make([]*Action, 0, 1)
+	err := requestBody.Board.InsertTileAt(requestBody.Direction, requestBody.Index)
+	if err == model.ErrUnsupportedDirection {
 		log.Printf("POST '/insert-tile' - Unsupported direction: %v", requestBody.Direction)
 		http.Error(w, fmt.Sprintf("unsupported direction: %v", requestBody.Direction), http.StatusInternalServerError)
-	}
-
-	actions := make([]*Action, 0, 1)
-	if err == nil {
+		return
+	} else if err == nil {
```

Comment:
What happen if the err is neither nil nor ErrUnsupportedDirection?

Reply:
It can also be `ErrInvalidAction` and `ErrEvenRow` which can be swallowed in the response